hera_pspec/pspecdata.py

```diff
@@ -810,35 +849,64 @@ def R(self, key):
             # in sqrt for some reason)
             sqrtT[np.isnan(sqrtT)] = 0.0
             sqrtY[np.isnan(sqrtY)] = 0.0
-
+            fext = self.filter_extension
+            #if we want to use a full-band filter, set the R-matrix to filter and then truncate.
+            tmat = np.zeros((self.spw_Nfreqs,
+                             self.spw_Nfreqs+np.sum(fext)),dtype=complex)
+            tmat[:,fext[0]:fext[0] + self.spw_Nfreqs] = np.identity(self.spw_Nfreqs,dtype=complex)
             # form R matrix
             if self.data_weighting == 'identity':
-                self._R[Rkey] = sqrtT.T * sqrtY.T * self.I(key) * sqrtY * sqrtT
+                self._R[Rkey] =  sqrtT.T ** 2. * np.dot(tmat, sqrtY.T * self.I(key) * sqrtY)
 
             elif self.data_weighting == 'iC':
-                self._R[Rkey] = sqrtT.T * sqrtY.T * self.iC(key) * sqrtY * sqrtT
+                self._R[Rkey] = sqrtT.T ** 2. * np.dot(tmat, sqrtY.T * self.iC(key) * sqrtY )
 
             elif self.data_weighting == 'sinc_downweight':
                 r_param_key = (self.data_weighting,) + key
                 if not r_param_key in self.r_params:
-                    raise ValueError("Error: no filter params specified for "
-                                    "sinc weights! ")
-                else:
-                    r_params = self.r_params[r_param_key]
-
+                    raise ValueError("r_param not set for %s!"%str(r_param_key))
+                r_params = self.r_params[r_param_key]
+                if not 'filter_centers' in r_params or\
+                   not 'filter_half_widths' in r_params or\
+                   not  'filter_factors' in r_params:
+                       raise ValueError("filtering parameters not specified!")
```

Comment:
regarding the value error, this is because the normalization factor should not be constant. fixed in #236

Reply:
For the example you give, what are the baselines and how is ds initialized? A script would be helpful.

Reply:
Sure. I have shared a notebook that details this.

Reply:
There are a few things going on. One issue (not the main one) is that the empirical covariance matrices for this data are poorly conditioned (cond >~ 1e18) so it's difficult to get a meaningful comparison between any two notebooks. 

There is a genuine change though that I've added that is being picked up. Namely, that I've moved the tapering to the left of iC so that 


R = T iC

in master we have

R = Sqrt(T) iC Sqrt(T)

which is effectively tapering the data before applying iC. I think the way I've changed it to is more correct since iC should be the covariance of the data its being applied too and moving a partial taper before iC changes the effective covariance of the data and makes iC less effective. In addition, since we have truncation as a possibility and the taper is meant to get desireable results from the DFT, I want the taper to be the same length as the truncated channels (which are being DFTd), not the length of the un-truncated channels.

This is a design change that I think merits some discussion before approval though. 

@acliu , @nkern , @philbull , do you have opinions on this?

-Aaron

Reply:
@philbull @saurabh-astro @nkern  @acliu . Last time we discussed, this it seemed that there was a concensus that moving the taper to after the foreground filter seemed the right thing to do but people wanted to figure out what the original thinking was as to when the taper was split up. Digging into the git history, the  tapering on either side of the foreground filter was added in https://github.com/HERA-Team/hera_pspec/commit/c0244f68ea0211180b40d7c34489015df996a8f8

Reply:
On the pspec telecon, we came to the following conclusion. 

We don't actually want to taper when we use inverse covariance weights so tapering is only really useful when what we call the K matrix is diagonal and square (identity weights with not tapering). Thus, 

* We are going to keep sqrt(T) I sqrt(T) for identity weights and any other weights that are diagonal. 
* We are not going to allow tapering when a non-diagonal R K matrix is used (this includes R-matrices that are truncated and R-matrices that subtract foregrounds). If users try to do this, we throw an error.

Reply:
After thinking about this a bit more, I am still not happy with sqrt(T) I sqrt(T) (if the weights are diagonal, this is equivalent to putting the taper on the left)

Tapering is something that you do when you are Fourier transforming. Thus, I really do think we should keep it next to where an FT step is (Q matrix). For example, one should be allowed to cook up a power spectrum estimator that applies non-diagonal weighting to the data (such as interpolation of flagged channels), truncates to a sub-band, and then performs a tapered Fourier transform and applies identity normalization. This soft of approach would not be possible with sqrt(T) I sqrt(T).